app/commands/v2/publish.rb

```diff
@@ -131,10 +130,10 @@ def publish_redirect_if_content_item_has_moved(new_location, previous_location,
         ) if draft_redirect
       end
 
-      def lookup_published_item(content_item)
+      def lookup_previous_item(content_item)
         ContentItemFilter.similar_to(
           content_item,
-          state: "published",
+          state: %w(published unpublished),
           base_path: nil,
           user_version: nil,
         ).first
```

Comment:
I think it would be better to go even further and make this a hard constraint of the system, i.e. that you can only have on published/unpublished content item in this history of a content_id.

There's a ticket here to address this: https://trello.com/c/EZXtVEpz/247-only-allow-one-of-unpublished-or-published-content-items

I think this would be a better approach because you'd check in a single place that this constraint has been met rather than repeatedly checking your assumptions and potentially littering the code.

I can add it now, but I think it would better to do the ticket linked above. What do you think?